clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -909,7 +937,7 @@ void CIRGenModule::buildGlobalVarDefinition(const clang::VarDecl *D,
   // from the type of the global (this happens with unions).
   if (!GV || GV.getSymType() != InitType) {
     // TODO(cir): this should include an address space check as well.
-    assert(0 && "not implemented");
+    setTypeOfGlobal(GV, InitType);
   }
```

Comment:
Oddly, this code path is not executed in the original codegen for the example in the PR description, despite the comment indicating otherwise.